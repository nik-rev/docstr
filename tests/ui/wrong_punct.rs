fn main() {
    // comma instead of ! is invalid
    docstr::docstr!(writeln,
        s,
        /// hello
        /// {}
        "world"
    );

    // missing comma
    docstr::docstr!(writeln!
        s
        /// hello
        /// {}
        "world"
    );

    // missing comma and no exclamation mark
    docstr::docstr!(writeln
        s
        /// hello
        /// {}
        "world"
    );

    // no exclamation mark
    docstr::docstr!(writeln
        s,
        /// hello
        /// {}
        "world"
    );
}
