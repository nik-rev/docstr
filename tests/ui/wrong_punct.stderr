error: replace with `!` to pass the macro: `writeln!`
 --> tests/ui/wrong_punct.rs:3:28
  |
3 |     docstr::docstr!(writeln,
  |                            ^

error: expected `,` after this
  --> tests/ui/wrong_punct.rs:12:9
   |
12 |         s
   |         ^

error: 2 identifiers in a row is not a valid macro path

       did you mean one of:
       - `writeln::s`
       - `writeln! s`
  --> tests/ui/wrong_punct.rs:20:9
   |
20 |         s
   |         ^

error: 2 identifiers in a row is not a valid macro path

       did you mean one of:
       - `writeln::s`
       - `writeln! s`
  --> tests/ui/wrong_punct.rs:28:9
   |
28 |         s,
   |         ^
